lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -442,17 +454,17 @@ function reduceUpdates(updates) {
 }
 
 function setupEventListeners(transport) {
-  function connect() {
-    transport._sendConnectOrSyncOrDisconnectMessage();
-  }
-
   function createOrResetTwilioConnection() {
     if (transport._twilioConnection) {
       transport._twilioConnection.removeListener('message', handleMessage);
     }
     const { _options, _wsServer } = transport;
-    const { TwilioConnection } = transport._options;
-    transport._twilioConnection = new TwilioConnection(_wsServer, _options);
+    const { TwilioConnection } = _options;
+
+    transport._twilioConnection = new TwilioConnection(_wsServer, Object.assign({
+      helloBody: transport._createConnectOrSyncOrDisconnectMessage()
```

Comment:
With the introduction of wait state, it could take long time for transport to actually send helloBody. I am wondering that helloBody should be obtained from the callback when transport is ready to send it. That way we can pass most recent offer/peer connection information.

Reply:
@makarandp0 I don't think that is necessary, because there is no way to change which tracks are published until the Room object is available to the developer. So, there is no way for the tracks in offer/answer to change until the Room is connected.